File: android/src/main/java/com/margelo/nitro/co/zyke/ble/BleNitroBleManagerFactory.kt
package com.margelo.nitro.co.zyke.ble

import com.margelo.nitro.core.*

/**
 * Android Factory implementation for BLE Nitro Manager
 * Creates BleNitroBleManager instances
 *
 * Note: restoreStateIdentifier and restoreStateCallback are iOS-specific
 * and ignored on Android
 */
class BleNitroBleManagerFactory : HybridNativeBleNitroFactorySpec() {

    override fun create(
        nativeRestoreStateIdentifier: String?,
        restoreStateCallback: ((peripherals: Array<BLEDevice>) -> Unit)?
    ): HybridNativeBleNitroSpec {
        // Ignore iOS-specific parameters on Android
        return BleNitroBleManager()
    }
}

File: packages/react-native-ble-nitro/ios/BleNitroBleManagerFactory.swift
import Foundation
import NitroModules

/**
 * iOS Factory implementation for BLE Nitro Manager
 * Creates BleNitroBleManager instances with optional state restoration configuration
 */
public class BleNitroBleManagerFactory: HybridNativeBleNitroFactorySpec_base, HybridNativeBleNitroFactorySpec_protocol {

    public func create(
        nativeRestoreStateIdentifier: String?,
        restoreStateCallback: (([BLEDevice]) -> Void)?
    ) throws -> any HybridNativeBleNitroSpec {
        return BleNitroBleManager(
            restoreStateIdentifier: nativeRestoreStateIdentifier,
            restoreStateCallback: restoreStateCallback
        )
    }
}
